website/docs/r/apprunner_service.html.markdown

```diff
@@ -236,7 +236,7 @@ The `code_configuration_values` blocks supports the following arguments:
 * `build_command` - (Optional) Command App Runner runs to build your application.
 * `port` - (Optional) Port that your application listens to in the container. Defaults to `"8080"`.
 * `runtime` - (Required) Runtime environment type for building and running an App Runner service. Represents a programming language runtime. Valid values: `PYTHON_3`, `NODEJS_12`, `NODEJS_14`, `NODEJS_16`, `CORRETTO_8`, `CORRETTO_11`, `GO_1`, `DOTNET_6`, `PHP_81`, `RUBY_31`.
-* `runtime_environment_secrets` - (Optional) Secrets and parameters available to your service as environment variables. A map of key/value pairs.
+* `runtime_environment_secrets` - (Optional) Secrets and parameters available to your service as environment variables. A map of key/value pairs, where the key is the desired name of the Secret in the environment (i.e. it does not have to match the name of the secret in Secrets Manager or SSM Parameter Store), and the value is the ARN of the secret from AWS Secrets Manager or the ARN of the parameter in AWS SSM Parameter Store.
```

Comment:
If you want to use a parameter from the AWS SSM Parameter Store, you need yo use the name of the parameter rather than the ARN.
The ARN is only used for secrets from the AWS Secrets Manager.
From what I understand, AWS App Runner determines whether it should use the AWS SSM Parameter Store or the AWS Secrets Manager based on whether you pass an ARN or a "name".

Reply:
@fracampit 👋 I referred to [this AWS document](https://docs.aws.amazon.com/apprunner/latest/dg/env-variable-manage.html#env-variable-manage.api) (specifically the second bullet, "Example Input.json file for image-based App Runner service") when I was reviewing, which seems to indicate that the SSM Parameter Store can be referenced by the ARN as well:

```json
"RuntimeEnvironmentSecrets": {
          "Credential1":"arn:aws:secretsmanager:<region>:<aws_account_id>:secret:XXXXXXXXXXXX",
          "Credential2":"arn:aws:ssm:<region>:<aws_account_id>:parameter/<parameter-name>"
        }
```

Now that you mention it, though, I do believe I may have read somewhere that Parameter Store parameter name could be used under certain circumstances. I'll look around a bit to see if I can find that information and raise a follow up PR if necessary.

Reply:
As it turns out, the bit I thought I'd recalled seeing was in the same document 😅:

> - If the SSM Parameter Store parameter is in the same AWS Region as the service that you want to launch, you can specify the full Amazon Resource Name (ARN) or the name of the parameter. If the parameter is in a different Region, you need to specify the full ARN.
> - Make sure that parameter that you're referencing to is in the same account as the service that you're launching or updating. Currently, you can't reference SSM Parameter Store parameter across accounts.

Reply:
Oh nice to know that, I was not aware!